cpp/Platform.Equality/Platform.Equality.Any.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#ifndef PLATFORM_EQUALITY_ANY_H
+#define PLATFORM_EQUALITY_ANY_H
+
+#include <any>
+#include <functional>
+#include <typeindex>
+#include <concepts>
+#include <stdexcept>
+
+namespace Platform::Equality
+{
+    template<class T>
+    inline auto __to_any_equal_visitor(auto func)
+    {
+        return std::pair<std::type_index, std::function<bool(std::any, std::any)>>
+        {
+                std::type_index(typeid(T)),
+                [func](std::any a, std::any b) -> std::size_t
+                {
+                    if constexpr (std::is_void_v<T>)
+                        return func();
+                    else
+                        return func(std::any_cast<T>(a), std::any_cast<T>(b));
+                }
+        };
+    }
+
+    #define BASE_VISITOR_REGISTER(Type) __to_any_equal_visitor<Type>([](Type a, Type b) {return a == b;})
+    #define VISITOR_REGISTER(Type, Func) __to_any_equal_visitor<Type>([](Type a, Type b) {return Func;})
+    static std::unordered_map<std::type_index, std::function<bool(std::any, std::any)>>
+    any_equal_visitor {
+            BASE_VISITOR_REGISTER(short int),
+            BASE_VISITOR_REGISTER(unsigned short int),
+            BASE_VISITOR_REGISTER(int),
+            BASE_VISITOR_REGISTER(unsigned int),
+            BASE_VISITOR_REGISTER(unsigned long int),
+            BASE_VISITOR_REGISTER(long long int),
+            BASE_VISITOR_REGISTER(unsigned long long int),
+            BASE_VISITOR_REGISTER(float),
+            BASE_VISITOR_REGISTER(double),
+            BASE_VISITOR_REGISTER(long double),
+
+            VISITOR_REGISTER(const char*, std::string(a) == std::string(b)),
+
+            BASE_VISITOR_REGISTER(const std::string&),
+    };
+    #undef BASE_VISITOR_REGISTER
+    #undef VISITOR_REGISTER
+
+    template<class T>
+    inline void register_any_equal_visitor(auto func)
```

Comment:
```cpp
template<class T>
inline void register_any_equal_visitor(T&& func) 
{
any_equal_visitor.insert(__to_any_equal_visitor<T>(std::forward<T>(func)));
}
```

Reply:
I can use the move idiom, but why add another template?

Reply:
One template for a type, another for functor. 
Instead of a template for a functor, you can use std::function
```cpp
template<class T>
inline void register_any_equal_visitor(std::function<bool(const T&, const&)> func) 
{
    any_equal_visitor.insert(__to_any_equal_visitor<T>(std::move(func)));
}
```

Reply:
Isn't that enough?
```cpp
requires (...) {func(a, b) -> std::same_as<bool>}
```

Reply:
🤔

Reply:
Yes, looks like this is enough